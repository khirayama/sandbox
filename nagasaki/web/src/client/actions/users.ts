import { Action } from 'redux';

export type Actions = SetUserName;

export interface SetUserName extends Action<'SET_USER_NAME'> {
  payload: {
    name: string;
  };
}

export const setUserName = (name: SetUserName['payload']['name']): SetUserName => ({
  type: 'SET_USER_NAME',
  payload: {
    name
  }
});
