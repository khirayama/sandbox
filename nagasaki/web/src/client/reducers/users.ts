import { Actions } from '../actions/users';

export interface State {
  name: string;
}

export const initialState: State = {
  name: ''
};

export const reducer = (state: State = initialState, action: Actions): State => {
  switch (action.type) {
    case 'SET_USER_NAME':
      return { ...state, name: action.payload.name };
    default:
      return state;
  }
};
