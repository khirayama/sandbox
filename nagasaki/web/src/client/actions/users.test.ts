import * as actions from './users';

test('should create an action to set username', () => {
  expect(actions.setUserName('foo')).toMatchSnapshot();
});
