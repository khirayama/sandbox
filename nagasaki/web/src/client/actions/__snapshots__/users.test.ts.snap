// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create an action to set username 1`] = `
Object {
  "payload": Object {
    "name": "foo",
  },
  "type": "SET_USER_NAME",
}
`;
